### DISCUSSION TITLE
is it possible to run on mobiles not chrome mobile

### DISCUSSION THREAD

**Discussion by tallevy22** [seleniumbase/SeleniumBase#1144] [Q&A] [answered]
Hi im working with Perfectomobile cloud for testing, to run on desktop I use these parameters: 
--browser=remote --server=XXX.perfectomobile.com/nexperience/perfectomobile --port=8080 --protocol=https --cap_file=../capabilities/cap_perfecto_chrome.py 
do I have an option to run on remote mobile?

**Comment by mdmintz** [comment, marked as answer]
Hi @tallevy22, the ``--mobile`` option enables the Chromium Mobile Device Emulator, which works for both Chrome browsers and Edge browsers. Here are the full details about it from the SeleniumBase docs:

--------

Use ``--mobile`` to quickly run your tests using Chrome's mobile device emulator with default values for device metrics (CSS Width, CSS Height, Pixel-Ratio) and a default value set for the user agent. To configure the mobile device metrics, use ``--metrics="CSS_Width,CSS_Height,Pixel_Ratio"`` to set those values. You'll also be able to set the user agent with ``--agent="USER-AGENT-STRING"`` (a default user agent will be used if not specified). To find real values for device metrics, [see this GitHub Gist](https://gist.github.com/sidferreira/3f5fad525e99b395d8bd882ee0fd9d00). For a list of available user agent strings, [check out this page](https://developers.whatismybrowser.com/useragents/explore/).

```bash
# Run tests using Chrome's mobile device emulator (default settings)
pytest test_swag_labs.py --mobile

# Run mobile tests specifying CSS Width, CSS Height, and Pixel-Ratio
pytest test_swag_labs.py --mobile --metrics="411,731,3"

# Run mobile tests specifying the user agent
pytest test_swag_labs.py --mobile --agent="Mozilla/5.0 (Linux; Android 9; Pixel 3 XL)"
```

--------

If you're running tests on a Selenium Grid that supports mobile emulation, you may need to pass that option as part of the capabilities file instead (if the ``--mobile`` option doesn't work). The https://github.com/seleniumbase/SeleniumBase/tree/master/examples/capabilities page has a ReadMe for using a capabilities file. The Perfectomobile website may have a list of custom capabilities that you can use. Each Grid provider has their own custom capabilities. First check if ``--mobile`` works, and if it doesn't, find the custom Perfectomobile option for the cap file instead.
